tests/unit/_modules/test_packagemanager.py

```diff
@@ -0,0 +1,345 @@
+import pytest
+from srv.salt._modules.packagemanager import PackageManager, Zypper, Apt
+from mock import MagicMock, patch, mock
+
+
+class TestPackageManager():
+    '''
+    This class contains a set of functions that test srv.salt._modules.packagemanager
+    '''
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_opensuse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('opensuse', 42.1, 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_suse(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('SUSE', '12.2', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Zypper) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_centos(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('centos', '8', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    def test_PackageManager_fedora(self, dist_return):
+        """
+        Test Packagemanager assignments based on `patform` mocks.
+        """
+        dist_return.return_value = ('fedora', '23', 'x86_64')
+        ret = PackageManager()
+        assert isinstance(ret.pm, Apt) is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_reboot(self, po, dist_return):
+        """
+        Reboot method
+        """
+        dist_return.return_value = ('opensuse', '42.2', 'x86_64')
+        ret = PackageManager()
+        ret.reboot_in()
+        assert po.called is True
+
+    @mock.patch('srv.salt._modules.packagemanager.linux_distribution')
+    @mock.patch('srv.salt._modules.packagemanager.Popen')
+    def test_not_implemented(self, po, dist_return):
+        """
+        Your platform is not supported
+        """
+        dist_return.return_value = ('ScientificLinux', '42.2', 'x86_64')
```

Comment:
I still think that checking for the existence of a package manager is the better option provided we check for zypper before apt. I think this will be able to support more systems with less code/work and will support the corner case where a distro changes from one package manager to another as Fedora has done more easily than hard-coding versions. I suspect this would make this particular code need less future maintenance. Do you think this logic holds up?

If this is what we choose, perhaps we should order them as below:
1) Check for zypper (before apt so we don't get a false positive with zypper-aptitude)
2) Check for dnf (before yum in case version of fedora have a yum wrapper for dnf, not yet implemented)
3) Check for yum (not yet implemented)
4) Check for apt (last in case there is a yum/dnf wrapper as with zypper)